app/src/org/commcare/android/database/DbHelper.java

```diff
@@ -77,7 +78,7 @@ public DbHelper(Context c) {
         return new Pair<String, String[]>(ret, arguments);
     }
     
-    public ContentValues getContentValues(Externalizable e) {
+    public ContentValues getContentValues(Externalizable e) throws StorageFullException {
```

Comment:
I'm not super clear on StorageFull here being the exception we throw, as we likely want to differentiate between a record being too large and the storage filling up.